### PR TITLE
vtk-m: update smoke test

### PR DESCRIPTION
**Pull request by vicentebolea**
This pull request updates the smoke_test to the current smoke test used in the vtk/vtk-m repo

### CODE REVIEW

**Comment by vicentebolea** [comment]
@wspear I updated this smoke test to match the one that we have at the VTK-m repo

**Comment by wspear** [comment]
Is this the same smoke test run by doing `spack test run vtk-m`? If so, there is an easier way to implement/maintain the test in the e4s testsuite. clean.sh and compile.sh are no longer needed and you just need run.sh to do:
```
#!/bin/bash
. ./setup.sh

spackTestRun $VTK-M_HASH
```
VTK-M_HASH is populated by spackLoadUnique vtk-m which is run in setup.sh.

**Comment by vicentebolea** [comment]
@wspear Thanks for the heads up.

will this still call `cmake -S src` ?

**Comment by wspear** [comment]
It will call
`spack test run vtk-m`
(But specifying a specific vtk-m hash)
And then determine success or failure and print the spack test run output. We're trying to move to this model for packages that have fully functional spack smoke tests so only one test needs to be maintained.

**Comment by vicentebolea** [comment]
I see, so we want to move to Spack smoke tests instead of having them copied here. I will figure out how to streamline our VTKm smoke test our Spack smoke test and make the changes here later on

**Comment by wspear** [comment]
Greetings @vicentebolea . Any news on updates to the vtk-m spack smoke test? It looks like you have a test method in the package but currently, it only runs on vtk-m@master and I don't think it would be found by a `spack test run` call.

**Comment by vicentebolea** [comment]
> Greetings @vicentebolea . Any news on updates to the vtk-m spack smoke test? It looks like you have a test method in the package but currently, it only runs on vtk-m@master and I don't think it would be found by a `spack test run` call.

What you have observed is correct, unfortunately no news yet, this is still on my backlog (although a bit deep inside), I expect it to be done in the coming two months.

**Comment by vicentebolea** [comment]
@wspear submitted PR: https://github.com/spack/spack/pull/29113

**Comment by wspear** [comment]
We can merge this once E4S's version of vtk-m hits 1.8.0-rc1+. Do you know when the official 1.8.0 release will arrive?

**Comment by vicentebolea** [comment]
@wspear sure

> Do you know when the official 1.8.0 release will arrive?

We are waiting for one last feature and one more short lived RC. Around 2 weeks.

**Comment by vicentebolea** [comment]
@wspear updated the vtk-m module to delegate in Spack

**Comment by vicentebolea** [comment]
ping @wspear

File: validation_tests/vtk-m/run.sh [hunk 1]
```diff
@@ -2,6 +2,4 @@
 
 source setup.sh
 
-set -x
-
-./build/smoke_test
+spackTestRun $VTK-M_HASH
```
Comment (wspear) [inline validation_tests/vtk-m/run.sh:5]: I'm not sure where VTK-M_HASH comes from for you but I don't think spack populates it when I load vtk-m and bash does not handle that string. 
```
==> Error: Unexpected token: '-'  Encountered when parsing spec:
    /-M_HASH
```

**Review by wspear — CHANGES_REQUESTED** [review]
I was planning to wait for the official 1.8 release but went ahead and tested with the rc.

File: validation_tests/vtk-m/run.sh (hunk 1 above)
Comment (vicentebolea) [reply validation_tests/vtk-m/run.sh]: apparently `setup.sh` does a tr '-' '_' for the PKG_HASH variable. I corrected the variable name in the last push

**Pull request merged**